### PR TITLE
DEP: misc: deprecate imported names

### PR DESCRIPTION
**Pull request by ev-br**
Scipy Roadmap says that scipy.misc is slated for removal at some point in time. The first step is then to deprecate the use of the aliases of functions which have been moved to more appropriate subpackages:

- comb, factorials (factorial, factorial2, factorialk) are in special,
- pade is in interpolate.

### CODE REVIEW

**Comment by rgommers** [comment]
+1

The refguide checker is unhappy

**Comment by ev-br** [comment]
Fixed the reST error, refguide-check seem happy now.

File: scipy/misc/__init__.py [hunk 1]
```diff
@@ -32,13 +28,25 @@
    imshow - Simple showing of an image through an external viewer [requires Pillow]
    info - Get help information for a function, class, or module
    lena - Get classic image processing example image Lena
+   toimage - Takes a numpy array and returns a PIL image [requires Pillow]
+   source - Print function source code
```
Comment (rgommers) [inline scipy/misc/__init__.py]: Why are these moved out of the deprecated functions list? They should be used as numpy functions, not scipy.misc aliases

Comment (ev-br) [inline scipy/misc/__init__.py]: Huh, it's just a GH way of showing the diff. I did not touch these ones. Do you want me to deprecate them as well in this PR?

Comment (rgommers) [inline scipy/misc/__init__.py]: Ah yes, misread - sorry about that. While you're at it, I think deprecating ``who`` and ``source`` definitely makes sense. They're just numpy aliases. 

``info`` has a slight modification, but I'm not sure it's of any use. It certainly doesn't belong anywhere other than ``misc``, so if we're getting rid of that.... and doesn't add much over ``help`` or if needed ``np.info(..., toplevel='scipy')``.

Comment (ev-br) [inline scipy/misc/__init__.py]: Agreed, `np.info` feels somehow in the same league as matplotlib's pylab: was possibly useful, not so much anymore. 

So I added deprecation warnings to all of `info`, `source` and `who`.

**Comment by rgommers** [comment]
LGTM, merged. Thanks Evgeni.
